### PR TITLE
Fixes JENKINS-59752 : add os ephemeral disk

### PR DESCRIPTION
**Pull request by bradleywangia**
- enable os ephemeral disks

### CODE REVIEW

**Comment by bradleywangia** [comment]
FIXES JENKINS-59752

**Comment by gavinfish** [comment]
Some useful links:
- https://docs.microsoft.com/en-us/azure/virtual-machines/windows/premium-storage-performance#disk-caching
- https://docs.microsoft.com/en-us/azure/virtual-machines/windows/ephemeral-os-disks
- https://docs.microsoft.com/en-us/azure/templates/microsoft.compute/2019-07-01/virtualmachines#DiffDiskSettings

File: src/main/java/com/microsoft/azure/vmagent/AzureVMManagementServiceDelegate.java [hunk 1]
```diff
@@ -214,8 +215,8 @@ public AzureVMDeploymentInfo createDeployment(
             }
             LOGGER.log(Level.INFO,
                     "AzureVMManagementServiceDelegate: createDeployment:"
-                            + " Creating a new deployment {0} with VM base name {1}",
-                    new Object[]{deploymentName, vmBaseName});
+                            + " Creating a new deployment {0} with VM base name {1}, ephemeralOSDisk={2}",
```
Comment (gavinfish) [inline src/main/java/com/microsoft/azure/vmagent/AzureVMManagementServiceDelegate.java]: I think this is not a good place to log ephemeral OS disk information.

File: src/main/resources/customImageTemplateWithManagedDisk.json [hunk 2]
```diff
@@ -28,6 +28,7 @@
                     "osDisk": {
                         "osState": "Generalized",
                         "osType": "[variables('osType')]",
+                        "caching": "[if(bool(variables('ephemeralOSDisk')), 'ReadOnly', json('null'))]",
```
Comment (gavinfish) [inline src/main/resources/customImageTemplateWithManagedDisk.json]: Why values are different between json files when `ephemeralOSDisk` is false?

File: src/main/resources/referenceImageIDTemplateWithManagedDisk.json [hunk 3]
```diff
@@ -88,6 +88,8 @@
                     },
                     "osDisk": {
                         "createOption": "FromImage",
+                        "caching": "[if(bool(variables('ephemeralOSDisk')), 'ReadOnly', json('null'))]",
+                        "diffDiskSettings": "[if(bool(variables('ephemeralOSDisk')), json('{\"option\": \"Local\"}'), json('null'))]",
```
Comment (gavinfish) [inline src/main/resources/referenceImageIDTemplateWithManagedDisk.json]: Only this json file has `diffDiskSettings` property?

File: src/main/java/com/microsoft/azure/vmagent/AzureVMManagementServiceDelegate.java (hunk 1 above)
Comment (bradleywangia) [reply src/main/java/com/microsoft/azure/vmagent/AzureVMManagementServiceDelegate.java]: removed ephemeral disk OS logging here

File: src/main/resources/referenceImageIDTemplateWithManagedDisk.json (hunk 3 above)
Comment (bradleywangia) [reply src/main/resources/referenceImageIDTemplateWithManagedDisk.json]: enabled this wherever ephemeral disk is in use per https://docs.microsoft.com/en-us/azure/templates/microsoft.compute/2019-07-01/virtualmachines#DiffDiskSettings

File: src/main/resources/customImageTemplateWithScriptAndManagedDisk.json [hunk 4]
```diff
@@ -35,6 +35,8 @@
                 "storageProfile": {
                     "osDisk": {
                         "osState": "Generalized",
+                        "caching": "[if(bool(variables('ephemeralOSDisk')), 'ReadOnly', json('null'))]",
+                        "diffDiskSettings": "[if(bool(variables('ephemeralOSDisk')), json('{\"option\": \"Local\"}'), json('null'))]",
```
Comment (gavinfish) [inline src/main/resources/customImageTemplateWithScriptAndManagedDisk.json]: Same one.

**Pull request merged**